library/math/count_paths/count_paths_rectangle.hpp

```diff
@@ -0,0 +1,37 @@
+/** @file */
+#pragma once
+#include "../../../kactl/content/numerical/NumberTheoreticTransform.h"
+#include "../n_choose_k/n_choose_k.hpp"
+/**
+ * @see https://noshi91.hatenablog.com/entry/2023/07/21/235339
+ * @code{.cpp}
+       auto [right, top] = get_right_and_top(left, bottom);
+ * @endcode
+ * @param left,bottom see image
+ * @returns [right, top] -> see image, note right.back() == top.back()
+ * @time O((n + m)log(n + m))
+ * @space O(n + m)
+ */
+array<vl, 2> get_right_and_top(vl left, vl bottom) {
+  assert(!empty(left) && !empty(bottom));
```

Comment:
one thing I always wonder is if we should be separating the asserts so that it's easy to see which part is breaking?